src/main/java/com/sap/sgs/phosphor/fosstars/data/github/NvdEntryMatcher.java

```diff
@@ -156,21 +174,111 @@ private boolean match(VendorData vendorData) {
   }
 
   /**
-   * Checks if product data from an NVD entry matches to the project's owner and name.
+   * Checks if product data from an NVD entry matches to the project's name.
    *
    * @param productData The produce data.
-   * @return True if the product data matches to the project's owner and name, false otherwise.
+   * @return True if the product data matches to the project's name, false otherwise.
    */
   private boolean match(ProductData productData) {
     return match(project.name(), productData.getProductName());
   }
 
   /**
-   * Returns true if one of two strings includes the other one.
+   * Checks if the a string is similar to another. It goes through a series of controlled checks to
+   * confirm this.
+   * 
+   * @param String one.
+   * @param String two.
+   * @return True if the accumulated points is greater than the threshold. Otherwise False.
+   */
+  private boolean match(String one, String two) {
+    if (one.equalsIgnoreCase(two)) {
+      return true;
+    }
+
+    int point = 0;
+
+    if (jaroWinklerSimilarityCheck(one, two)) {
+      point += 1;
+    }
+
+    if (longestCommonSubsequenceCheck(one, two)) {
+      point += 1;
+    }
+
+    return point > 0 && point / 2 > METRIC_THRESHOLD;
+  }
+
+  /**
+   * Find the Longest Common Subsequence found between the 2 strings. And check if the subsequence
+   * score is not lesser than a certain threshold of the first string.
+   * 
+   * @param String one.
+   * @param String two.
+   * @return True if subsequence score is not lesser than the threshold, Otherwise False.
+   * @see <a href="https://www.geeksforgeeks.org/longest-common-subsequence-dp-4/">Longest Common
+   *      Subsequence Problem</a>
+   */
+  private static boolean longestCommonSubsequenceCheck(String one, String two) {
+    return LCS.apply(one, two) >= (one.length() * METRIC_THRESHOLD);
+  }
+
+  /**
+   * Find the Jaro Winkler Similarity score between the 2 strings. And check if the score is not
+   * lesser than a certain threshold.
+   * 
+   * @param String one.
+   * @param String two.
+   * @return True if the score is not lesser than the threshold, Otherwise False.
+   * @see <a href="https://www.geeksforgeeks.org/jaro-and-jaro-winkler-similarity/">Jaro Winkler
+   *      Similarity</a>
+   */
+  private static boolean jaroWinklerSimilarityCheck(String one, String two) {
+    return JWS.apply(one, two) > METRIC_THRESHOLD;
+  }
+
+  /**
+   * Checks if one of the cpeXXUri contains the project's name.
+   * 
+   * @param cpeMatch of type {@link CpeMatch}.
+   * @return True if one of the cpeXXUri contains the project's name. Otherwise False.
+   */
+  private boolean cpeUriCheck(CpeMatch cpeMatch) {
+    if (StringUtils.isNotEmpty(cpeMatch.getCpe23Uri())) {
+      return match(productFrom23Uri(cpeMatch.getCpe23Uri()));
+    } else if (StringUtils.isNotEmpty(cpeMatch.getCpe22Uri())) {
+      return match(productFrom22Uri(cpeMatch.getCpe22Uri()));
+    }
+    return false;
```

Comment:
How about this:

```java
return match(productFrom23Uri(cpeMatch.getCpe23Uri()))
    || match(productFrom22Uri(cpeMatch.getCpe22Uri()))
```

`stringAtIndex()` will then need to return an empty string if `content` is null.